### PR TITLE
[CBRD-24795] When using a function under <= and >= conditions, the plan generator does not perform a range scan

### PR DESCRIPTION
**Pull request by youngjinj**
http://jira.cubrid.org/browse/CBRD-24795

Make the function expression used in the function index the target in the process of rewriting the query below.
1. qo_converse_sarg_terms
   - convert terms of the form 'const op attr' to 'attr op const'
   ```
   (-) attribute1 operation (-) attribute2 → attribute1 converse_operation attribute2
   [ x ] (-) attribute1 operation attribute2 → attribute1 converse_operation (-) attribute2
   (-) attribute1 operation constant → attribute1 converse_operation (-) constant
   constant operation (-) attribute1 → (-) constant converse_operation attribute1
   
   /* swap */
   attribute2 #1 operation attribute1 #3 → attribute1 #3 converse_operation attribute2 #1
   constant operation attribute1 → attribute1 converse_operation constant
   ```
2. qo_reduce_comp_pair_terms (qo_search_comp_pair_term)
   - reduce a pair of comparison terms into one BETWEEN term
   ```
   attribute operation const1 AND attribute operation const2
    → attribute BETWEEN const1 PT_BETWEEN_operation> const2
   ```
3. qo_convert_to_range (qo_convert_to_range_helper, qo_merge_range_helper)
   - convert comparison terms to RANGE
   ```
   attribute1 operation constant1 OR attribute1 operation constant2
   → attribute1 RANGE (constant1 operation gt_inf max OR min inf_lt constant2)
   
   attribute1 BETWEEN constant1 gt_lt constant2
   → attribute1 RANGE (constant1 gt_lt constant2)
   
   attribute1 IN (constant1, constant2, constant3);
   → attribute1 RANGE (constant1 = OR constant2 = OR constant3 =)
   
   /* merge */
   (attribute1 operation constant1 OR attribute1 operation constant3) AND (attribute1 operation constant2 OR attribute1 operation constant3)
   → (attribute1 operation constant1) AND (attribute1 operation constant2 OR attribute1 operation constant3)
   ```
4. qo_apply_range_intersection (qo_apply_range_intersection_helper)
   - narrow search range by applying range intersection
   ```
   /* merge */
   attribute1 RANGE (constant1 operation constant2) AND attribute1 RANGE (constant3 = OR constant4 = OR constant5 =)
   → attribute1 RANGE (constant3 = OR constant4 = OR constant5 =)
   ```
5. qo_do_auto_parameterize
   - auto-parameterize convert value in expression to host variable (input marker)

### CODE REVIEW

File: src/parser/semantic_check.c [hunk 1]
```diff
@@ -13788,7 +13788,54 @@ pt_check_path_eq (PARSER_CONTEXT * parser, const PT_NODE * p, const PT_NODE * q)
...
+      {
+	const char *p_str = NULL;
+	const char *q_str = NULL;
+	unsigned int save_custom;
+
+	save_custom = parser->custom_print;	/* save */
+	parser->custom_print |= PT_CONVERT_RANGE;
+
+	// p_str = (const char *) pt_get_varchar_bytes (pt_print_bytes (parser, p));
+	// q_str = (const char *) pt_get_varchar_bytes (pt_print_bytes (parser, q));
```
Comment (shparkcubrid) [inline src/parser/semantic_check.c]: Plz remove this.

Comment (youngjinj) [reply src/parser/semantic_check.c]: Thank you. I will remove it.

File: src/parser/semantic_check.c [hunk 2]
```diff
@@ -13761,7 +13761,7 @@ pt_check_path_eq (PARSER_CONTEXT * parser, const PT_NODE * p, const PT_NODE * q)
 	{
 	  return 1;
 	}
-      break;
```
Comment (shparkcubrid) [inline src/parser/semantic_check.c:13825]: Not related to the issue. How about putting the following condition first?
```
if (p->info.name.spec_id != q->info.name.spec_id)
	{
	  return 1;
	}
```

File: src/parser/semantic_check.c [hunk 3]
```diff
@@ -13840,7 +13887,31 @@ pt_check_class_eq (PARSER_CONTEXT * parser, PT_NODE * p, PT_NODE * q)
...
```
Comment (shparkcubrid) [inline src/parser/semantic_check.c:13904]: Not related to the issue. How about putting the following condition first?
```
if (p->info.name.spec_id != q->info.name.spec_id)
	{
	  return 1;
	}
```

**Review by shparkcubrid — APPROVED** [review]

File: src/parser/semantic_check.c (hunk 2 above)
Comment (youngjinj) [reply src/parser/semantic_check.c:13825]: I agree. I will change it.

File: src/parser/semantic_check.c (hunk 3 above)
Comment (youngjinj) [reply src/parser/semantic_check.c:13904]: I agree. I will change it.

**Comment by youngjinj** [comment]
There are no changes to failing tests, except for the 2 below.
These 2 fail for other reasons unrelated to this issue.
- sql/_24_aprium_qa/_02_sql_extension/issue_9413_trunc_date/cases/trunc_date_ps.sql
- sql/_24_aprium_qa/_02_sql_extension/issue_9488_round_date/cases/round_date_ps.sql

**Review by xmilex-git — APPROVED** [review]

**Review by beyondykk9 — APPROVED** [review]

**Pull request merged**